services/app-web/src/index.tsx

```diff
@@ -62,7 +62,24 @@ Amplify.configure({
 
 const authMode = process.env.REACT_APP_AUTH_MODE
 assertIsAuthMode(authMode)
-const cache = new InMemoryCache()
+const cache = new InMemoryCache({
+    typePolicies: {
+        ContractRevision: {
+            fields: {
+                formData: {
+                    merge: true,
+                },
+            },
+        },
+        RateRevision: {
+            fields: {
+                formData: {
+                    merge: true,
```

Comment:
Getting GQL warnings (not blocking but they clutter the console) because form datas dont have `id` but are nested objects that are changing. 

https://www.apollographql.com/docs/react/caching/cache-field-behavior/#merging-non-normalized-objects 

Following docs, I believe this is the proper way to address but welcome other eyes.

Reply:
interesting, funny because these aren't really stand alone objects, we just made formData to wrap the data up nicely. we could give them a fake ID that's the same as the enclosing revision if that helps. I will read more about the merging stuff